### DISCUSSION TITLE
Helix does not use Kitty Keyboard Protocol through WSL on Windows 10 Wezterm

### DISCUSSION THREAD

**Discussion by Sarund9** [helix-editor/helix#11810] [Troubleshooting]
### Summary

Helix does not receive Enhanced Keyboard support from Wezterm on windows while on WSL.

The Windows version (EXE) of Helix works on Wezterm when:
- The Wezterm config `allow_win32_input_mode` is Enabled
- Not running through WSL

Helix on Windows also works when running through Windows Terminal

Running the windows executable through WSL does not work, presumably because WSL discards win32 inputs before reaching Helix.
Neither the WSL or Native windows version of WSL is receiving Enhanced Kitty Keyboard Inputs.

Some questions:
1) How can I verify that the current shell, is capable of receiving extended keyboard input ?
2) How can I run Helix to verify that it's actually requesting the extended protocols ?

Notes:
- I have also installed Helix with `brew` and it does not work either.

### Reproduction Steps

Run Wezterm on Windows with the config options:
```lua
local wezterm = require("wezterm")

local config = wezterm.config_builder()

config.enable_kitty_keyboard = true
config.allow_win32_input_mode = false
config.enable_csi_u_key_encoding = false

 -- Making sure Wezterm does not intercept SHIFT+Backspace
config.keys = {
   {
        key = 'Backspace',
        mods = 'SHIFT',
        action = wezterm.action.DisableDefaultAssignment,
    }
}
```

Log into WSL (debian) and run Helix.

Helix Config:
```toml
[keys.normal]
S-backspace = 'command_pallete'
```

The Keybind does not work.

### Helix log

Log is empty

### Platform

Windows 10 - Debian WSL

### Terminal Emulator

Wezterm 20240203.110809.0

### Installation Method

Releases Page

### Helix Version

helix 24.7 (079f5442)

**Comment by the-mikedavis** [comment]
You can run Helix with `hx -vv` to enable debug logging. Then you should see a line like this if Helix can detect that the kitty keyboard protocol is available:

```
2024-09-23T18:04:16.024 helix_tui::backend::crossterm [DEBUG] The keyboard enhancement protocol is supported in this terminal (checked in 3.670335ms)
```

**Comment by Huliiiiii** [comment]
This might be related to https://github.com/wez/wezterm/issues/1673#issuecomment-1054311400.
I use wezterm ssh to wsl to solve this problem.
